File: Downsampling.m
% Read the image
image = imread('Azadi_Square.jpg');

% Downsample by a factor of 1/2
downsampled_image_half = downsample(image, 2);

% Downsample by a factor of 1/4
downsampled_image_quarter = downsample(image, 4);

% Display the original and downsampled images
subplot(1, 3, 1);
imshow(image);
title('Original Image');

subplot(1, 3, 2);
imshow(downsampled_image_half);
title('Downsampled (1/2)');

subplot(1, 3, 3);
imshow(downsampled_image_quarter);
title('Downsampled (1/4)');

File: Python/Downsampling.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:


import cv2
import numpy as np

# Read the image
image = cv2.imread('Azadi_Square.jpg')

# Downsample by a factor of 1/2
downsampled_image_half = cv2.resize(image, None, fx=0.5, fy=0.5, interpolation=cv2.INTER_NEAREST)

# Downsample by a factor of 1/4
downsampled_image_quarter = cv2.resize(image, None, fx=0.25, fy=0.25, interpolation=cv2.INTER_NEAREST)

# Display the original and downsampled images
cv2.imshow('Original Image', image)
cv2.imshow('Downsampled (1/2)', downsampled_image_half)
cv2.imshow('Downsampled (1/4)', downsampled_image_quarter)
cv2.waitKey(0)
cv2.destroyAllWindows()

